File: Data Structures & Algorithms/duplicate-integer/submission-1.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        HashSet<int> seen = new HashSet<int>();
        for (int i = 0; i < nums.Length; i+=1) {
            if (seen.Contains(nums[i])) {
                return true;
            }
            seen.Add(nums[i]);
        }
        return false;
    }
}

File: Data Structures & Algorithms/non-cyclical-number/submission-1.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n != 1:
            sum_digits = 0
            while n:
                digit = n % 10
                digit = digit ** 2
                sum_digits += digit
                n = n // 10
            if sum_digits in seen:
                return False
            else:
                seen.add(sum_digits)
                n = sum_digits
        return True